src/midiEditorCommands.cpp

```diff
@@ -257,15 +339,19 @@ class MidiEventIterator {
 		return this->index <= other.index;
 	}
 
-	const double operator[](const int index) const {
-		double pos = this->getEvent(this->index + index);
-		return MIDI_GetProjTimeFromPPQPos(take, pos);
+	value_type operator[](const difference_type index) const{
+		return this->getEvent(this->index + index);
 	}
 
-	const double operator*() const {
+	reference operator*() const {
 		return (*this)[0];
 	}
 
+	pointer operator->() const {
+		this->currentValue = (*this)[0];
```

Comment:
Ug. I'm kinda regretting suggesting you use `->` now. I didn't realise it'd be this painful or ugly. I honestly thought it'd be the same as `(*foo).bar` at the end of the day, but apparently not. Anyway, it's done now and I think it probably does make things more readable in future.